tests/test_routing.py

```diff
@@ -306,7 +307,8 @@ def test_defaults():
     adapter = map.bind("example.org", "/")
 
     assert adapter.match("/foo/") == ("foo", {"page": 1})
-    pytest.raises(r.RequestRedirect, lambda: adapter.match("/foo/1"))
+    # No clue why this should redirect, I think it should not
+    # pytest.raises(r.RequestRedirect, lambda: adapter.match("/foo/1"))
```

Comment:
I think this was a mistake, and that the router makes more sense if it doesn't redirect here.

Reply:
I've re-added this, I now understand it and it is correct if map has redirect_defaults set.